### PR TITLE
Core/BattlePets: Misc fixes

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Define BattlePetBreedQuality enum class.
- Check the quality of the battle pet species in battle_pet_quality table to avoid invalid values.
-  Set CurrentBattlePetBreedQuality and WildBattlePetLevel update fields with the data of the summoned battle pet.
-  Added function to calculate the WildBattlePetLevel of wild battle pets according to their spawn zone.

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/BattlePets/BattlePetMgr.cpp [hunk 1]
```diff
@@ -169,7 +180,7 @@ uint8 BattlePetMgr::GetDefaultPetQuality(uint32 species)
 {
     auto itr = _defaultQualityPerSpecies.find(species);
     if (itr == _defaultQualityPerSpecies.end())
-        return 0; // default poor
+        return AsUnderlyingType(BattlePetBreedQuality::Poor); // Default
```
Comment (Shauren) [inline src/server/game/BattlePets/BattlePetMgr.cpp:183]: Why don't you make this return `BattlePetBreedQuality` instead?

File: src/server/game/Entities/Creature/Creature.cpp [hunk 2]
```diff
@@ -1620,6 +1620,16 @@ void Creature::UpdateLevelDependantStats()
     SetStatFlatModifier(UNIT_MOD_ARMOR, BASE_VALUE, armor);
 }
 
+void Creature::SelectWildBattlePetLevel()
+{
+    if (IsWildBattlePet())
+    {
+        AreaTableEntry const* areaTable = sAreaTableStore.AssertEntry(GetZoneId());
```
Comment (Shauren) [inline src/server/game/Entities/Creature/Creature.cpp:1627]: Don't assert area, check for null

Comment (meji46) [reply src/server/game/Entities/Creature/Creature.cpp]: Fixed. Tested on GM Island

File: src/server/game/BattlePets/BattlePetMgr.cpp (hunk 1 above)
Comment (meji46) [reply src/server/game/BattlePets/BattlePetMgr.cpp]: I've changed the return type to BattlePetBreedQuality. I don't know if you want me to extend it to other fields like WorldPackets::BattlePet::BattlePet.Quality

**Pull request merged**